File: KnowledgeAssistant.Domain/Documents/Document.cs
namespace KnowledgeAssistant.Domain.Documents
{
    public class Document
    {
        public int Id { get; set; }

        public string Title { get; set; } = string.Empty;

        public string OriginalText { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; }

        public IEnumerable<string> Topics { get; set; } = Array.Empty<string>();
    }
}

File: KnowledgeAssistant.Web/src/app/models/document.ts
export interface DocumentItem {
  id: number;
  title: string;
  originalText: string;
  createdAt: string;
  topics: string[];
}

export interface Topic {
  id: number;
  name: string;
  parentId: number | null;
}

/**
 * Global chunking/embedding settings, mirroring the WPF "Retrieval Settings" panel which reads/writes
 * `/api/configuration/chunking-settings` (not the per-document retrieval-config endpoint).
 */
export interface ChunkingSettings {
  embeddingModelName: string;
  chunkTargetSizeChars: number;
  chunkOverlapChars: number;
}

export const DEFAULT_CHUNKING_SETTINGS: ChunkingSettings = {
  embeddingModelName: '',
  chunkTargetSizeChars: 1200,
  chunkOverlapChars: 200,
};